include/greentea-client/test_env.h

```diff
@@ -80,24 +80,144 @@ extern const char *GREENTEA_TEST_ENV_LCOV_START;
+ */
 void GREENTEA_SETUP_UUID(const int timeout, const char *host_test_name, char *buffer, size_t size);
-void GREENTEA_TESTSUITE_RESULT(const int);
+
+/**
+ * Notify the host (__exit message) side that test suite execution was complete.
+ *
+ * @note If __exit is not received by the host side we will assume TIMEOUT.
+ *
+ * @param result Test suite result
+ */
+void GREENTEA_TESTSUITE_RESULT(const int result);
+
+/**
+ *  Test Case support
+ */
+
+/**
+ * Notify the host side that test case started.
+ *
+ * @param test_case_name Test case name
+ */
 void GREENTEA_TESTCASE_START(const char *test_case_name);
-void GREENTEA_TESTCASE_FINISH(const char *test_case_name, const size_t passes, const size_t failed);
+
+/**
+ * Notify host side that test case finished.
+ *
+ * @param test_case_name Test case name
+ * @param passes Number of test passes
+ * @param failures Number of test failures
+ */
+void GREENTEA_TESTCASE_FINISH(const char *test_case_name, const size_t passes, const size_t failures);
 
 /**
  *  Test suite result related notification API
  */
-void greentea_send_kv(const char *, const int);
-void greentea_send_kv(const char *, const int, const int);
-void greentea_send_kv(const char *, const char *, const int);
-void greentea_send_kv(const char *, const char *, const int, const int);
+
+/**
+ * Encapsulate and send key-value message from DUT (device under test) to host
+ *
+ * @details This function uses underlying functions to write directly
+ *          to the serial port, (CONSOLE_TX). This allows KVs to be used
+ *          from within interrupt context.
+ *
+ * @param key Message key (message/event name)
+ * @param value Message payload, integer value
+ */
+void greentea_send_kv(const char *key, const int value);
+
+/**
+ * Encapsulate and send key-value-value message from DUT (device under test) to host
+ *
+ * @details This function uses underlying functions to write directly
+ *          to the serial port, (CONSOLE_TX). This allows key-value-value to be used
+ *          from within interrupt context.
+ *
+ * @note Names of the parameters: This function is used to send numbers
+ *       of passes and failures to host. But it can be used to send any
+ *       key-value-value (string-integer-integer) message to host.
+ *
+ * @param key Message key (message/event name)
+ * @param passes Send additional integer formatted data
+ * @param failures Send additional integer formatted data
+ *
+ */
+void greentea_send_kv(const char *key, const int passes, const int failures);
+
+/**
+ * Encapsulate and send key-value-value message from DUT (device under test) to host
+ *
+ * @details This function uses underlying functions to write directly
+ *          to the serial port, (CONSOLE_TX). This allows KVs to be used
+ *          from within interrupt context.
+ *
+ * @note Names of the parameters: This function is used to send numbers
+ *       of passes and failures to host. But it can be used to send any
```

Comment:
This function doesn't seem to send "numbers of passes and failures to the host", it only takes a single integer parameter named `result`. We don't seem to explain what `result` represents or how it is interpreted by the host tools.

Reply:
Looks like this override (string, string, int) isn't used by utest, so no clue what `result` represents 😟

Reply:
Interesting, I guess we can update the comment when we find out what it is. Or, if it's not used anywhere, maybe removing it is best? The current state should be OK for this PR anyway.